python/ray/serve/_private/long_poll.py

```diff
@@ -247,24 +249,29 @@ async def listen_for_change(
         immediately if the snapshot_ids are outdated, otherwise it will block
         until there's an update.
         """
-        watched_keys = keys_to_snapshot_ids.keys()
-        existent_keys = set(watched_keys).intersection(set(self.snapshot_ids.keys()))
-
         # If there are any keys with outdated snapshot ids,
         # return their updated values immediately.
-        updated_objects = {
-            key: UpdatedObject(self.object_snapshots[key], self.snapshot_ids[key])
-            for key in existent_keys
-            if self.snapshot_ids[key] != keys_to_snapshot_ids[key]
-        }
+        updated_objects = {}
+        for key, snapshot_id in keys_to_snapshot_ids.items():
```

Comment:
Can we rename `snapshot_id` to `client_snapshot_id` for clarity?

```suggestion
        for key, client_snapshot_id in keys_to_snapshot_ids.items():
```